pkg/controller/certificates/certificate_controller.go

```diff
@@ -196,21 +196,27 @@ func (cc *CertificateController) maybeSignCertificate(key string) error {
 		if err != nil {
 			return fmt.Errorf("error auto approving csr: %v", err)
 		}
+		_, err = cc.kubeClient.Certificates().CertificateSigningRequests().UpdateApproval(csr)
+		if err != nil {
+			return fmt.Errorf("error updating approval for csr: %v", err)
+		}
 	}
 
 	// At this point, the controller needs to:
 	// 1. Check the approval conditions
 	// 2. Generate a signed certificate
 	// 3. Update the Status subresource
 
-	if csr.Status.Certificate == nil && IsCertificateRequestApproved(csr) {
-		certBytes, err := cc.signer.Sign(csr)
+	if cc.signer != nil && csr.Status.Certificate == nil && IsCertificateRequestApproved(csr) {
+		csr, err := cc.signer.Sign(csr)
+		if err != nil {
+			return fmt.Errorf("error auto signing csr: %v", err)
```

Comment:
No, I don't think so. I added a comment about AutoApprove and Signer interfaces:

```
+// err returned from these interfaces should indicate utter failure that
 +// should be retried. "Buisness logic" errors should be indicated by adding
 +// a condition to the CSRs status, not by returning an error.
```